src/server/scripts/Battlegrounds/TempleOfKotmogu/battleground_temple_of_kotmogu.cpp

```diff
@@ -0,0 +1,630 @@
+class spell_bg_temple_of_kotmogu_victory_points_periodic : public SpellScript
+{
+public:
+    virtual void OnHit(SpellEffIndex /*effIndex*/) const
+    {
+        Player* player = GetHitPlayer();
+        if (!player)
+            return;
+
+        Battleground* bg = player->GetBattleground();
+        if (!bg)
+            return;
+
+        uint32 const maxScore = sWorldStateMgr->GetValue(TempleOfKotmogu::WorldStates::MaxPoints, player->GetMap());
+        bg->UpdatePvpStat(player, TempleOfKotmogu::PvpStats::VictoryPoints, Points);
+        Team const team = player->GetBGTeam();
+        uint32 const score = bg->GetTeamScore(GetTeamIdForTeam(team));
+
+        int32 const amount = std::min(maxScore, score + Points);
+        bg->SetTeamPoint(team, amount);
+        sWorldStateMgr->SetValue(TempleOfKotmogu::WorldStates::AlliancePoints, bg->GetTeamScore(TEAM_ALLIANCE), false, player->GetMap());
+        sWorldStateMgr->SetValue(TempleOfKotmogu::WorldStates::HordePoints, bg->GetTeamScore(TEAM_HORDE), false, player->GetMap());
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_bg_temple_of_kotmogu_victory_points_periodic::OnHit, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 112910 - +10 Victory Points!
+template<int32 Points>
+class spell_bg_temple_of_kotmogu_victory_points_kill : public spell_bg_temple_of_kotmogu_victory_points_periodic<Points>
+{
+    void OnHit(SpellEffIndex effIndex) const override
+    {
+        spell_bg_temple_of_kotmogu_victory_points_periodic<Points>::OnHit(effIndex);
+        Player* player = SpellScript::GetHitPlayer();
+        if (!player)
+            return;
+
+        if (ZoneScript* zonescript = player->FindZoneScript())
+            zonescript->DoAction(TempleOfKotmogu::Actions::CheckScore, player, player);
+    }
+
+    void Register() override
+    {
+        SpellScript::OnEffectHitTarget += SpellScript::SpellEffectFn(spell_bg_temple_of_kotmogu_victory_points_kill::OnHit, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+struct at_bg_temple_of_kotmogu_healing_buff : AreaTriggerAI
+{
+    explicit at_bg_temple_of_kotmogu_healing_buff(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) { }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        if (Player* player = unit->ToPlayer())
+            if (ZoneScript* zonescript = at->GetZoneScript())
+                zonescript->DoAction(TempleOfKotmogu::Actions::ConsumeAreaTrigger, at, player);
+    }
+};
+
+struct at_bg_temple_of_kotmogu_small_area : AreaTriggerAI
+{
+    explicit at_bg_temple_of_kotmogu_small_area(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) { }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        if (Player* player = unit->ToPlayer())
+        {
+            if (!_totalTimeInAreaWithOrb.contains(player->GetGUID()))
+                _totalTimeInAreaWithOrb[player->GetGUID()] = 0;
+
+            player->CastSpell(player, TempleOfKotmogu::Spells::SmallAura, true);
```

Comment:
I know passing true (aka full trigger) is the easiest way, but it's preferable to pass only required cast flags to mimic the behavior if possible.